libraries/standard/mqtt/src/mqtt.c

```diff
@@ -601,30 +818,124 @@ static MQTTStatus_t handleIncomingPublish( MQTTContext_t * const pContext,
     LogInfo( ( "De-serialized incoming PUBLISH packet: DeserializerResult=%d", status ) );
 
     if( status == MQTTSuccess )
+    {
+        /* Check if dup bit is set by the broker. Duplicates are not possible
+         * for QoS0 publishes. */
+        if( ( publishInfo.dup ) && ( ( publishInfo.qos == MQTTQoS1 ) ||
+                                     ( publishInfo.qos == MQTTQoS2 ) ) )
```

Comment:
```suggestion
        if( ( publishInfo.dup ) && ( publishInfo.qos > MQTTQoS0 ) )
```

has less conditions and would be easier for unit test coverage

Reply:
yea. Agree. Thanks